tools/configure/configure.ml

```diff
@@ -1236,28 +564,57 @@ let write_makefile f =
   pr "COQWARNERROR=%s\n" (if prefs.warn_error then "-w +default" else "");
   pr "CONFIGURE_DPROFILE=%s\n" prefs.dune_profile;
   pr "COQ_INSTALL_ENABLED=%b\n" prefs.install_enabled;
-  if !dune_29 then pr "DUNE_29_PLUS=yes\n";
-  close_out o;
-  Unix.chmod f 0o444
-
-let _ = write_makefile "config/Makefile"
+  if dune_29 then pr "DUNE_29_PLUS=yes\n";
+  ()
 
-let write_dune_c_flags f =
-  safe_remove f;
-  let o = open_out f in
+let write_dune_c_flags cflags o =
   let pr s = fprintf o s in
-  pr "(%s)\n" cflags;
-  close_out o;
-  Unix.chmod f 0o444
-
-let _ = write_dune_c_flags "config/dune.c_flags"
+  pr "(%s)\n" cflags
 
-let write_configpy f =
-  safe_remove f;
-  let o = open_out f in
+let write_configpy o =
   let pr s = fprintf o s in
   pr "# DO NOT EDIT THIS FILE: automatically generated by ../configure\n";
   pr "version = '%s'\n" coq_version;
   pr "is_a_released_version = %s\n" (if is_a_released_version then "True" else "False")
 
-let _ = write_configpy "config/coq_config.py"
+(* Main configure routine *)
+let main () =
```

Comment:
Maybe some comments here explaining what is going on roughly?

Reply:
I guess it is self-explanatory, comments here would feel a bit redundant; what part you think is unclear?

Reply:
Actually, on second reading I think it is quite clear.

Reply:
Actually, on second reading it is quite clear.